src/Shared/StackTrace/ExceptionDetails/LoggerExtensions.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.StackTrace.Sources
+{
+    internal static class LoggerExtensions
+    {
+        private static readonly Action<ILogger, Exception> _failedToReadStackFrameInfo;
+
+        static LoggerExtensions()
+        {
+            _failedToReadStackFrameInfo = LoggerMessage.Define(
+                logLevel: LogLevel.Debug,
+                eventId: new EventId(1, "FailedToReadStackTraceInfo"),
```

Comment:
Is this `EventId` OK? I'm not sure how it works when you don't have control over the (name of the) logger that's passed in.

Reply:
Then use 0.

Reply:
It can still clash and that's the problem with defining logging outside of the component. It needs to be owned by the component that does the logging. So it needs to be a parameter...

Reply:
0 is fine though, we currently use that when we don't specify the event id